lib/file_transfer_agent/azure_util.js

```diff
@@ -236,6 +236,65 @@ function azure_util(azure, filestream)
     meta['resultStatus'] = resultStatus.UPLOADED;
   }
 
+  /**
+  * Download the file blob then write the file.
+  *
+  * @param {String} dataFile
+  * @param {Object} meta
+  * @param {Object} encryptionMetadata
+  * @param {Number} maxConcurrency
+  *
+  * @returns {null}
+  */
+  this.nativeDownloadFile = async function (meta, fullDstPath, maxConcurrency)
+  {
+    var stageInfo = meta['stageInfo'];
+    var client = this.createClient(stageInfo);
+    var azureLocation = this.extractContainerNameAndPath(stageInfo['location']);
+    var blobName = azureLocation.path + meta['srcFileName'];
+
+    var containerClient = client.getContainerClient(azureLocation.containerName);
+    var blockBlobClient = containerClient.getBlockBlobClient(blobName);
+
+    try
+    {
+      var downloadBlockBlobResponse  = await blockBlobClient.download(0);
+      var readableStream = downloadBlockBlobResponse.readableStreamBody;
+
+      await new Promise((resolve, reject) =>
+      {
+        var writer = fs.createWriteStream(fullDstPath);
```

Comment:
Does this create the destination path if it doesn't exist? If not then we should be creating this directory before trying to download any files

Reply:
The folder is created before in file_transfer_agent.js
https://github.com/snowflakedb/snowflake-connector-nodejs/blob/ba5859a994757e46fbbeb5faeef5c10a6db8762d/lib/file_transfer_agent/file_transfer_agent.js#L118-L123